### PR TITLE
[feat] filter nodes

### PR DESCRIPTION
**Pull request by Adityapnn811**
- Able to filter nodes by typing the id and separating it by comma
- Filter button is implemented as performance might get ugly if it always change on input

### CODE REVIEW

File: svelte/src/routes/components/scripts/helper/index.ts [hunk 1]
```diff
@@ -22,21 +22,6 @@ export function extractAvailableEdgeType(links: GraphDataEdge[]) {
 	return availableEdgeType;
 }
 
-// Currently, this function combine the weights with undeleted duplicated links in mind
```
Comment (christojeffrey) [inline svelte/src/routes/components/scripts/helper/index.ts:25]: have you check that this function is not used by anyone?

File: svelte/src/scripts/draw/filter-node.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+				filteredNodeNames.push(member.id);
+				populateFilterId(member.members, filteredNodeNames);
+			});
+			return false;
+		}
+		return true;
+	});
+
+	for (let i = 0; i < filteredNodes.length; i++) {
+		filteredNodes[i].members = filteredNodes[i].originalMembers;
```
Comment (christojeffrey) [inline svelte/src/scripts/draw/filter-node.ts]: this means that when a node is filtered and then removed, they will always be uncollapsed, right?

**Review by christojeffrey — COMMENTED** [review]
do you think this should be put on filter instead of draw?

File: svelte/src/routes/components/scripts/helper/index.ts (hunk 1 above)
Comment (Adityapnn811) [reply svelte/src/routes/components/scripts/helper/index.ts:25]: yes

File: svelte/src/scripts/draw/filter-node.ts (hunk 2 above)
Comment (Adityapnn811) [reply svelte/src/scripts/draw/filter-node.ts]: I'm not really sure what you mean by always be uncollapsed

**Comment by Adityapnn811** [comment]
> do you think this should be put on filter instead of draw?

I did think about it, but I backed out. Now I think it makes more sense in refilter, there is also an unwanted behavior onCollapse if I put it in redraw because the link will be redirected first and won't be filtered out correctly.

**Comment by Adityapnn811** [comment]
@christojeffrey I have fixed it using try catch. It will recalculate using another algorithm first, then, recompute using the user's settings algorithm.

**Review by christojeffrey — APPROVED** [review]
not sure but fix later I suppose :v

**Pull request merged**